diff --git a/mbed-os/targets/TARGET_STM/TARGET_STM32H7/TARGET_STM32H747xI/TARGET_STM32H747xI_CM7/TOOLCHAIN_GCC_ARM/STM32H747xI_CM7.ld b/mbed-os/targets/TARGET_STM/TARGET_STM32H7/TARGET_STM32H747xI/TARGET_STM32H747xI_CM7/TOOLCHAIN_GCC_ARM/STM32H747xI_CM7.ld
index 68280f4..414a952 100644
--- a/mbed-os/targets/TARGET_STM/TARGET_STM32H7/TARGET_STM32H747xI/TARGET_STM32H747xI_CM7/TOOLCHAIN_GCC_ARM/STM32H747xI_CM7.ld
+++ b/mbed-os/targets/TARGET_STM/TARGET_STM32H7/TARGET_STM32H747xI/TARGET_STM32H747xI_CM7/TOOLCHAIN_GCC_ARM/STM32H747xI_CM7.ld
@@ -76,6 +76,17 @@ ENTRY(Reset_Handler)
 
 SECTIONS
 {
+    .tcm.text :
+    {
+        *(.tcm.text*)
+    } > ITCMRAM
+
+    .tcm.bss (NOLOAD) :
+    {
+        . = ALIGN(8);
+        *(.tcm.bss*)
+    } > DTCMRAM
+
     .text :
     {
         KEEP(*(.isr_vector))
@@ -145,7 +156,6 @@ SECTIONS
         KEEP(*(.fini_array))
         PROVIDE_HIDDEN (__fini_array_end = .);
 
-        KEEP(*(.jcr*))
         . = ALIGN(8);
         /* All data end */
         __data_end__ = .;
@@ -153,19 +163,6 @@ SECTIONS
 
     } > RAM
 
-    /* Uninitialized data section
-     * This region is not initialized by the C/C++ library and can be used to
-     * store state across soft reboots. */
-    .uninitialized (NOLOAD):
-    {
-        . = ALIGN(32);
-        __uninitialized_start = .;
-        *(.uninitialized)
-        KEEP(*(.keep.uninitialized))
-        . = ALIGN(32);
-        __uninitialized_end = .;
-    } > RAM
-
     .bss :
     {
         . = ALIGN(8);
